### PR TITLE
nrf_modem: update build system files

### PR DESCRIPTION
**Pull request by lemrey**
Introduce a Kconfig choice for the binary version.

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: nrf_modem/Kconfig [hunk 1]
```diff
@@ -12,14 +15,24 @@ choice NRF_MODEM_BUILD_STRATEGY
...
+	default NRF_MODEM_BINARY_DEFAULT
+
+config NRF_MODEM_BINARY_DEFAULT
+	bool "default"
 	help
-	  Links the application with the library version capable of emitting logs.
-	  This increases the final size of the application.
+	  Link with the default binary.
+
+config NRF_MODEM_BINARY_LOG
```
Comment (eivindj-nordic) [inline nrf_modem/Kconfig:31]: I'm wondering whether we should keep NRF_MODEM_LOG to enable modem library logs(?). This changes the Kconfig option to enable library logs and requires an update to the documentation.

Comment (eivindj-nordic) [reply nrf_modem/Kconfig:31]: Would it makes sense to keep NRF_MODEM_LOG and have that select the binary version?

Comment (lemrey) [reply nrf_modem/Kconfig:31]: Yes, it would. That's a good point, I will change that.

**Comment by lemrey** [comment]
Have changed the commits, please revist.

File: nrf_modem/Kconfig [hunk 2]
```diff
@@ -1,10 +1,15 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
```
Comment (eivindj-nordic) [inline nrf_modem/Kconfig:2]: nit: This file is older than this

**Review by eivindj-nordic — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**